packages/yew/src/functional/hooks/mod.rs

```diff
@@ -33,6 +33,10 @@ pub fn use_hook<InternalHook: 'static, Output, Tear: FnOnce(&mut InternalHook) +
     runner: impl FnOnce(&mut InternalHook, HookUpdater) -> Output,
     destructor: Tear,
 ) -> Output {
+    if !CURRENT_HOOK.is_set() {
+        wasm_bindgen::throw_str("Hooks can only be used in the scope of a function component");
```

Comment:
Any reason we should throw a JavaScript exception in this case instead of panicking?

Reply:
When I tried panic the error message was lost :(

Reply:
Interesting... Did you use the `console.error` panic hook?

Reply:
Thank you! 
🤦 Silly me, I tested it by throwing a hook in to main before the app started, hence the panic hook wasn't set.